src/Ghcid.hs

```diff
@@ -156,7 +159,18 @@ autoOptions o@Options{..}
                                 "stack exec --test --bench -- ghci" : opts
                 in f flags $ stack:cabal
               | ".ghci" `elem` files -> f ("ghci":opts) [curdir </> ".ghci"]
-              | cabal /= [] -> f (if null arguments then "cabal repl":map ("--ghc-options=" ++) opts else "cabal exec -- ghci":opts) cabal
+              | cabal /= [] ->
+                  let useCabal =
+                          let prefix =
+                                  [ "cabal", "repl" ] ++ Foldable.toList target
```

Comment:
I'd put this on one one. Better to use `maybeToList` rather than the more general Foldable.